compaction/task_manager_module.hh

```diff
@@ -48,23 +50,36 @@ protected:
 
 class major_compaction_task_impl : public compaction_task_impl {
 public:
+    enum class flush_mode {
+        unset,
```

Comment:
Is this a meaningful option compared to `skip`? What will the behaviour be when this is `unset`.

Reply:
This is for backward compatibility with internal paths that generate a `major_compaction_task_impl`.
I think it would be safe to have an optional flush_mode there and to set it by default to `compacted_tables` which is the preexisting behavior.

Reply:
`unset` is removed in next version.
the base class sets the default to `compacted_tables` (though it's unused when it's constructure as a base class of `major_compaction_task_executor`)